File: python/2015/005/part_one.py
import common


def is_nice(test):
    forbidden = ["ab", "cd", "pq", "xy"]
    for nope in forbidden:
        if nope in test:
            return False

    vowels = [char for char in test if char in "aeiouAEIOU"]
    if len(vowels) < 3:
        return False

    has_dupe = False
    for a in range(len(test) - 1):
        if test[a] == test[a + 1]:
            has_dupe = True
            break

    if not has_dupe:
        return False

    return True


if __name__ == "__main__":
    for input_file in common.inputs:
        nices = 0
        for line in common.read_file(input_file):
            if is_nice(line):
                nices += 1

        print(nices)

File: kotlin/src/main/kotlin/pt/dmb/adventofcode/y2015/p05/PartOne.kt
package pt.dmb.adventofcode.y2015.p05

import pt.dmb.adventofcode.common.Utilities

class PartOne {
    private var nices = 0

    fun isNice(test: String): Boolean {
        val forbidden = listOf("ab", "cd", "pq", "xy")
        forbidden.forEach { nope ->
            if (test.contains(nope)) return false
        }

        val vowels = test.filter { it in "aeiouAEIOU" }
        if (vowels.length < 3) return false

        var hasDupe = false
        for (a in 0 until test.length - 1) {
            if (test[a] == test[a + 1]) {
                hasDupe = true
                break
            }
        }
        if (!hasDupe) return false

        return true
    }

    fun increaseNices() {
        nices++
    }

    fun countNices() = nices
}

fun main() {
    val solver = PartOne()
    Utilities.readInputFile(2015, 5, scope = "main").forEach { line ->
        if (solver.isNice(line)) {
            solver.increaseNices()
        }
    }
    println(solver.countNices())
}